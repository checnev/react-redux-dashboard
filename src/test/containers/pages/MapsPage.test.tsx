import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import state from 'test/data/initialState';

import MapsPageContainer from 'containers/pages/Maps';


jest.mock('react-yandex-maps', () => ({
  YMaps: (props: any) => <div data-testid="YMaps" {...props}></div>,
  Map: (props: any) => {
    const { defaultState, instanceRef, ...otherProps } = props;
    return <div data-testid="Map" {...otherProps}></div>
  },
  Placemark: (props: any) => <div data-testid="Placemark" {...props}></div>,
}));



const mockStore = configureMockStore([thunk]);
let store: any = null;

let container: any = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.append(container);

  store = mockStore(state);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('MapsPageContainer', () => {

  const setUp = (component: React.ReactNode) => (
    <Provider store={store}>
      <Router>
        {component}
      </Router>
    </Provider>
  );

  it('MapsPageContainer render', async () => {
    await act(async () => {
      render(setUp(<MapsPageContainer />), container);
    });

    expect(
      pretty(container.innerHTML)
    ).toMatchSnapshot();
  });

});
