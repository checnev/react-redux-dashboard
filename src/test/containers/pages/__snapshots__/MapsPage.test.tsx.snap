// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapsPageContainer MapsPageContainer render 1`] = `
"<div class=\\"MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-3\\">
  <div class=\\"MuiGrid-root makeStyles-header-2 MuiGrid-item MuiGrid-grid-xs-12\\">
    <h1 class=\\"MuiTypography-root MuiTypography-h4\\">Автомобили на карте</h1>
  </div>
  <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\">
    <hr class=\\"MuiDivider-root\\">
  </div>
  <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\">
    <div data-testid=\\"YMaps\\">
      <div data-testid=\\"Map\\" width=\\"100%\\" height=\\"350\\">
        <div data-testid=\\"Placemark\\" geometry=\\"1,1\\" properties=\\"[object Object]\\" modules=\\"geoObject.addon.balloon\\" options=\\"[object Object]\\"></div>
        <div data-testid=\\"Placemark\\" geometry=\\"1,1\\" properties=\\"[object Object]\\" modules=\\"geoObject.addon.balloon\\" options=\\"[object Object]\\"></div>
      </div>
    </div>
  </div>
</div>"
`;
